src/containers/App.js

```diff
@@ -2,32 +2,18 @@ import { PropTypes, Component } from 'react';
 import Radium from 'radium';
 import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
 import getMuiTheme from 'material-ui/styles/getMuiTheme';
-import HeaderBar from './HeaderBar';
-
-const inlineStyles = {
-  routerContainer: {
-    maxHeight: 'calc(100vh - 64px)',
-    overflowY: 'auto',
-  },
-};
 
 @Radium
 class App extends Component {
   static propTypes = {
     children: PropTypes.node,
-    location: PropTypes.object,
   }
 
   render() {
     return (
       <div>
         <MuiThemeProvider muiTheme={getMuiTheme()}>
-          <div>
-            <HeaderBar location={this.props.location} />
```

Comment:
Yes. I find this better than adding hacks to the HeaderBar itself. (By "hack" I mean checking if the user is logged in.)

Basically, we have only one page with a header bar now. Furthermore, it is not reusable as you're unlikely to want a search bar and an import/export menu on other pages. (That's why I would further inline it in the ArticlesPage.)